### PR TITLE
issue 68: processing all patches for nginx from openresty

### LINKED ISSUE #68: Performance warning: ngx.sleep(0) called without delayed events patch
Hello,

I'm encountering a performance warning in the logs of my nginx container which I believe could be addressed by applying the delayed events patch to the ngx_lua module. The warning is as follows:

2023/07/27 13:33:09 [warn] 39#39: *14672 ngx.sleep(0) called without delayed events patch, this will hurt performance, client: 127.0.0.1, server: , request: "GET /metrics HTTP/1.1", host: "localhost:9145"

From what I understand, the ngx.sleep() function is being called in a blocking manner, potentially leading to performance issues in high-concurrency scenarios. The delayed events patch would change the behavior of ngx.sleep() to be non-blocking, which would allow the worker process to continue processing other requests while the sleep time is ongoing.

I'm building my container based on an existing image that already contains the built Lua components, so I'm unable to apply this patch myself during the build process.

Given the potential performance implications, I think it would be beneficial if the delayed events patch could be included in future versions of the base image.

Thank you for considering this request.

### PR DESCRIPTION
**Pull request by fabiocicerchia**
closes #68

### CODE REVIEW

**Review by alcneo — APPROVED** [review]

**Comment by alcneo** [comment]
Seems there are still no patches for the relevant version. Newest available seems to be for nginx-1.23.0 - https://github.com/openresty/openresty/blob/master/patches/nginx-1.23.0-delayed_posted_events.patch

File: nginx/1.25.1/alpine/3.18.2/tpl/Makefile [hunk 1]
```diff
@@ -19,19 +23,22 @@ core:
...
 	mkdir -p /var/cache/nginx/client_temp \
 		/var/cache/nginx/proxy_temp \
 		/var/cache/nginx/fastcgi_temp \
 		/var/cache/nginx/uwsgi_temp \
-		/var/cache/nginx/scgi_temp
+		/var/cache/nginx/scgi_temp \
+		/nginx-${VER_NGINX}/patches
+	git clone https://github.com/openresty/openresty.git
+	PATCH_VER=${VER_NGINX}; \
+	if [ "$(shell ls -1 openresty/patches/nginx-${VER_NGINX}-*.patch)" = "" ]; then \
```
Comment (alcneo) [inline nginx/1.25.1/alpine/3.18.2/tpl/Makefile:36]: Potential issue with expanding the wildcard, might be a useful workaround iterate with a loop and.

	PATCH_VER=${VER_NGINX}
	if [ "$(ls -1 openresty/patches/nginx-${VER_NGINX}-*.patch 2>/dev/null)" = "" ]; then
	    PATCH_VER=${FALLBACK_VER_NGINX}
	fi
	for file in openresty/patches/nginx-${PATCH_VER}-*.patch
	do
	    cp "$file" /nginx-${VER_NGINX}/patches/
	done

File: tpl/patches/nginx-1.23.0-no_pool.patch [hunk 2]
```diff
@@ -0,0 +1 @@
+# emptying the openresty patch because not compatible with nginx 1.25.1
```
Comment (fabiocicerchia) [inline tpl/patches/nginx-1.23.0-no_pool.patch:1]:
```
#9 52.48 applying openresty patch openresty/patches/nginx-1.23.0-no_pool.patch
#9 52.48 patching file src/core/nginx.h
#9 52.48 Hunk #1 FAILED at 10.
#9 52.48 1 out of 1 hunk FAILED -- saving rejects to file src/core/nginx.h.rej
#9 52.48 patching file src/core/ngx_array.c
#9 52.48 patching file src/core/ngx_array.h
#9 52.48 patching file src/core/ngx_palloc.c
#9 52.48 patching file src/core/ngx_palloc.h
```

File: tpl/patches/nginx-1.23.0-safe_resolver_ipv6_option.patch [hunk 3]
```diff
@@ -0,0 +1 @@
+# emptying the openresty patch because not compatible with nginx 1.25.1
```
Comment (fabiocicerchia) [inline tpl/patches/nginx-1.23.0-safe_resolver_ipv6_option.patch:1]:
```
#9 52.49 applying openresty patch openresty/patches/nginx-1.23.0-safe_resolver_ipv6_option.patch
#9 52.49 patching file src/core/ngx_resolver.c
#9 52.49 Hunk #1 FAILED at 426.
#9 52.49 Hunk #2 succeeded at 459 (offset 16 lines).
#9 52.49 1 out of 2 hunks FAILED -- saving rejects to file src/core/ngx_resolver.c.rej
```

File: tpl/patches/nginx-1.23.0-server_header.patch [hunk 4]
```diff
@@ -0,0 +1 @@
+# emptying the openresty patch because not compatible with nginx 1.25.1
```
Comment (fabiocicerchia) [inline tpl/patches/nginx-1.23.0-server_header.patch:1]:
```
#9 52.49 applying openresty patch openresty/patches/nginx-1.23.0-server_header.patch
#9 52.49 patching file src/core/nginx.h
#9 52.49 Hunk #1 FAILED at 11.
#9 52.49 1 out of 1 hunk FAILED -- saving rejects to file src/core/nginx.h.rej
#9 52.49 patching file src/http/ngx_http_header_filter_module.c
#9 52.49 patching file src/http/v2/ngx_http_v2_filter_module.c
#9 52.49 Hunk #1 succeeded at 115 (offset -28 lines).
```

File: tpl/patches/nginx-1.23.0-setting_args_invalidates_uri.patch [hunk 5]
```diff
@@ -0,0 +1 @@
+# emptying the openresty patch because not compatible with nginx 1.25.1
```
Comment (fabiocicerchia) [inline tpl/patches/nginx-1.23.0-setting_args_invalidates_uri.patch:1]:
```
#9 52.49 applying openresty patch openresty/patches/nginx-1.23.0-setting_args_invalidates_uri.patch
#9 52.49 patching file src/http/ngx_http_variables.c
#9 52.49 Hunk #1 FAILED at 15.
#9 52.49 Hunk #2 FAILED at 218.
#9 52.49 Hunk #3 succeeded at 771 with fuzz 1 (offset 124 lines).
#9 52.50 2 out of 3 hunks FAILED -- saving rejects to file src/http/ngx_http_variables.c.rej
```

File: tpl/patches/nginx-1.23.0-ssl_client_hello_cb_yield.patch [hunk 6]
```diff
@@ -0,0 +1 @@
+# emptying the openresty patch because not compatible with nginx 1.25.1
```
Comment (fabiocicerchia) [inline tpl/patches/nginx-1.23.0-ssl_client_hello_cb_yield.patch:1]:
```
#9 52.50 applying openresty patch openresty/patches/nginx-1.23.0-ssl_client_hello_cb_yield.patch
#9 52.50 patching file src/event/ngx_event_openssl.c
#9 52.50 Hunk #1 FAILED at 1712.
#9 52.50 Hunk #2 succeeded at 1894 (offset 5 lines).
#9 52.50 1 out of 2 hunks FAILED -- saving rejects to file src/event/ngx_event_openssl.c.rej
```

File: tpl/patches/nginx-1.23.0-stream_proxy_timeout_fields.patch [hunk 7]
```diff
@@ -0,0 +1 @@
+# emptying the openresty patch because not compatible with nginx 1.25.1
```
Comment (fabiocicerchia) [inline tpl/patches/nginx-1.23.0-stream_proxy_timeout_fields.patch:1]:
```
#9 52.51 applying openresty patch openresty/patches/nginx-1.23.0-stream_proxy_timeout_fields.patch
#9 52.51 patching file src/stream/ngx_stream.h
#9 52.51 Hunk #1 succeeded at 248 (offset -6 lines).
#9 52.51 Hunk #2 succeeded at 310 (offset -6 lines).
#9 52.51 patching file src/stream/ngx_stream_proxy_module.c
#9 52.51 Hunk #1 succeeded at 402 (offset 2 lines).
#9 52.51 Hunk #2 succeeded at 411 (offset 2 lines).
#9 52.51 Hunk #3 succeeded at 713 (offset 2 lines).
#9 52.51 Hunk #4 succeeded at 721 (offset 2 lines).
#9 52.51 Hunk #5 succeeded at 791 (offset 2 lines).
#9 52.51 Hunk #6 FAILED at 963.
#9 52.51 Hunk #7 succeeded at 995 (offset 4 lines).
#9 52.51 Hunk #8 succeeded at 1070 (offset 4 lines).
#9 52.51 Hunk #9 succeeded at 1121 (offset 6 lines).
#9 52.51 Hunk #10 succeeded at 1430 (offset 6 lines).
#9 52.51 Hunk #11 succeeded at 1442 (offset 6 lines).
#9 52.51 Hunk #12 succeeded at 1463 (offset 6 lines).
#9 52.51 Hunk #13 succeeded at 1625 (offset 6 lines).
#9 52.51 Hunk #14 succeeded at 1819 (offset -10 lines).
#9 52.51 1 out of 14 hunks FAILED -- saving rejects to file src/stream/ngx_stream_proxy_module.c.rej
```

**Comment by fabiocicerchia** [comment]
@alcneo I'll need some more time to patch some of the patches to be compatible with nginx 1.25.1

**Pull request merged**